src/Microsoft.OData.Client/BaseEntityType.cs

```diff
@@ -16,7 +16,7 @@ public class BaseEntityType : IBaseEntityType
         /// </summary>
         protected internal DataServiceContext Context { get; set; }
 
-        DataServiceContext IBaseEntityType.DataServiceContext
+        DataServiceContext IBaseEntityType.EntityTrackingContext

```

Comment:
The source class should also probably change to `EntityTrackingContext` as well.

Reply:
I think changing the actual class name from `DataServiceContext` to `EntityTrackingContext` would be a much bigger change. The `DataServiceContext` also does more than entity tracking, it is basically the entry point to the client and can be used even when entity tracking is disabled. I think renaming that class to something else should be a larger discussion.

Reply:
Interesting. In that case, I think the property name should change to `DataServiceContext` to match the class name. Sorry about that, I should have dug deeper.

It does seem like the DataServiceContext class could use a solid streamlining.